src/thinblock.cpp

```diff
@@ -247,6 +248,107 @@ bool CXThinBlock::CheckBlockHeader(const CBlockHeader &block, CValidationState &
     return true;
 }
 
+bool CXThinBlock::HandleMessage(CDataStream &vRecv, CNode *pfrom, string strCommand, unsigned nHops)
+{
+    if (!pfrom->ThinBlockCapable())
+    {
+        LOCK(cs_main);
+        Misbehaving(pfrom->GetId(), 100);
+        return error("%s message received from a non thinblock node, peer=%d", strCommand, pfrom->GetId());
+    }
+
+    int nSizeThinBlock = vRecv.size();
+    CXThinBlock thinBlock;
+    vRecv >> thinBlock;
+
+    // Message consistency checking
+    if (!IsThinBlockValid(pfrom, thinBlock.vMissingTx, thinBlock.header))
+    {
+        LOCK(cs_main);
+        Misbehaving(pfrom->GetId(), 100);
+        return error("Invalid %s received", strCommand);
+    }
+
+    // Is there a previous block or header to connect with?
+    {
+        LOCK(cs_main);
+        uint256 prevHash = thinBlock.header.hashPrevBlock;
+        BlockMap::iterator mi = mapBlockIndex.find(prevHash);
+        if (mi == mapBlockIndex.end())
+        {
+            Misbehaving(pfrom->GetId(), 10);
+            return error("%s from peer %s (%d) will not connect, unknown previous block %s",
+                         strCommand, pfrom->addrName.c_str(), pfrom->id, prevHash.ToString());
```

Comment:
this one of the site blamed to be not properly formatted by `clang-format-3.8`, this is the actual diff: 

```diff
--- thinblock.cpp
+++ thinblock.cpp_formatted
@@ -277,8 +277,8 @@
         if (mi == mapBlockIndex.end())
         {
             Misbehaving(pfrom->GetId(), 10);
-            return error("%s from peer %s (%d) will not connect, unknown previous block %s",
-                         strCommand, pfrom->addrName.c_str(), pfrom->id, prevHash.ToString());
+            return error("%s from peer %s (%d) will not connect, unknown previous block %s", strCommand,
+                pfrom->addrName.c_str(), pfrom->id, prevHash.ToString());
```